### ISSUE TITLE
first rotamer is being skipped

### ISSUE THREAD

**Issue by orbeckst** [MDAnalysis/RotamerConvolveMD#6]
In the discussion for PR #3  it became clear that the existing code has a bug: the first rotamer in the library is never included in the calculation.

**Comment by orbeckst** [comment]
Actually, I am confused as to what is in the library of rotamers: 

The DCD file with conformers has 214 frames but the dat file with the weights only contains 213 entries:

``` python
>>> import MDAnalysis as mda
>>> u = mda.Universe("./rotamer1_R1A_298K.pdb", "./rotamer1_R1A_298K.dcd")
>>> print(u.trajectory.n_frames)
214
>>> print(len(open("./R1A_298K_populations.dat").readlines()))
213
```

In the code, this discrepancy in length was taken care of by **prepending a weight of 0** to the array of weights in [rotcon/library.py L115](https://github.com/MDAnalysis/RotamerConvolveMD/blob/master/rotcon/library.py#L115). (In @philipwfowler 's original code, [`[0, 0]` was prepended to the weights](https://github.com/MDAnalysis/RotamerConvolveMD/blob/f5975e89271dbbdee3d7f65879e0334e251f2d1c/convolve-mtss-rotatmers.py#L17) – the first 0 to account for MDAnalysis's 1-based frames (prior to release 0.11.0), the second one to pad the weights array and to ignore the first conformer in the library.)

@philipwfowler can you clarify what the first frame in the DCD is supposed to be?
- Is it correct to not include it (in which case this issue is not referring to a bug, just an implementation detail that is porrly documented) or
- is one weight missing (in which case this is certainly some sort of bug). 

What do Gunnar Jeschke's original files look like?

**Comment by philipwfowler** [comment]
I am confused too but am also in Boston so will have a look when I get back. I think I am going to have to go back to the original communications with Gunnar Jeschke.

—Phil

**Comment by philipwfowler** [comment]
Ok, (big sigh), this is also a bug. Gunnar sent me PDB (213) and an XYZ file (213 frames) and I converted it into a DCD file. If you look at the DCD file, frames #0 and #1 are the same, hence there are 214 frames, which is incorrect. 

So I have removed the identical frame #0 resulting in this DCD which has 213 frames, hence matching the number of weights. There are definitely 213 weights because I didn’t alter the file and they sum to unity.

Please could you add this new DCD file?

Yours,

—Phil

**Comment by orbeckst** [comment]
I'll do it. 

But it's not a bug because we weighted the first frame with 0 so no double counting. 

All previous results are valid. 

## 

Oliver Beckstein
email: orbeckst@gmail.com

Am May 20, 2016 um 4:42 schrieb Philip W Fowler notifications@github.com:

> Ok, (big sigh), this is also a bug. Gunnar sent me PDB (213) and an XYZ file (213 frames) and I converted it into a DCD file. If you look at the DCD file, frames #0 and #1 are the same, hence there are 214 frames, which is incorrect. 
> 
> So I have removed the identical frame #0 resulting in this DCD which has 213 frames, hence matching the number of weights. There are definitely 213 weights because I didn’t alter the file and they sum to unity.
> 
> Please could you add this new DCD file?
> 
> Yours,
> 
> —Phil
>

**Comment by orbeckst** [comment]
I'll create a new DCD from the old one by just deleting the first frame:

``` python
import MDAnalysis as mda
u = mda.Universe("./rotamer1_R1A_298K.pdb", "./rotamer1_R1A_298K.dcd")
with mda.Writer("./new_rotamer1_R1A_298K.dcd", n_atoms=u.atoms.n_atoms) as W:
    for ts in u.trajectory[1:]:
        W.write(ts)
%mv new_rotamer1_R1A_298K.dcd ./rotamer1_R1A_298K.dcd
```

Because of MDAnalysis/mdanalysis#859 I can't create the TRR at the moment (but that one isn't used). I will delete the TRR for the time being.

**Comment by orbeckst** [comment]
Fixed by 2e32d0b
